hack/benchmark/src/singleflight/singleflight_bench_test.go

```diff
@@ -0,0 +1,283 @@
+//
```

Comment:
should we put this bench file under `internal/singleflight` ?

Reply:
@kpango what do you think?

Reply:
@rinx sorry I forgot to tag you.

Reply:
I do not think move bench file to another dir.

Reply:
so should we separate the bench code of this improved implementation and the standard implementation?
(if we do so please remember to leave a comment to the test code for users to compare the difference of performance)

Reply:
i misunderstood.
it will depend on our rule. i think we have 2 option.
1. create code bench file under the `hack` dir. and we keep to organize dir as same as Vald directory construction.
2. create code bench file under the same dir like test code.

Is there anything option? and  what do you think about the above options?
@kevindiu @hlts2

Reply:
I think :
1. separate the bench code of current implement and standard implementation (as it takes 30mins to run in my PC)
2. put the bench code of our implementation to the same directory of source code
3. add comment to our bench code to inform user that they can also execute the bench code for the standard package and compare the result

Reply:
I'd like to confirm `1`.
what do you mean the bench code of current implement and standard implementation?

Reply:
the current implementation of the bench test includes 2 different bench test, one is for the current implementation (@hlts2 improved) and the second one is the golang standard implementation. We want to compare the performance difference of the improved one and the golang standard one.

`1` means separate them and put them in different package

Reply:
hmm.
@hlts2 could you please tell us your opinion?

Reply:
:memo:
we discussed with the author.
And It was decided that it should be as it is.